==========================================
Interactive Tests for ``integer_validator()``
==========================================

	>>> from '0-add_integer' import add_integer  # doctest: +SKIP
	>>> cls = __import__('7-base_geometry').BaseGeometry


	>>> data = {\
	... 		'int': [0, 1, -1, 534, -435, 123456789, -123456789],\
	... 		'float': [0, 1.7, -1.7, 56.12345, -56.123456, 0.123456789, -0.123456789],\
	... 		'str': ["", "ta	b", "spa ce", "nospace", "@#$_&", "12ty"],\
	... 		'list': [[], [1, 2.6, "string", {}, set(), (), None]],\
	... 		'dict': [{}, {"k1": 1, "k2": None}],\
	... 		'set': [set(), {'k1', 'k2', 1, 2.7, ()}],\
	... 		'tuple': [(), (1, 2.9, "string", {}, None)],\
	... 		'none': [None]}

	>>> n = data['int']

	>>> fl = data['float']

	>>> s = data['str']

	>>> l = data['list']

	>>> d = data['dict']

	>>> st = data['set']

	>>> t = data['tuple']


	>>> bg = cls()
	>>> bg.integer_validator("my_int", 12)
	>>> bg.integer_validator("name", "John")
	Traceback (innermost last):
	TypeError: name must be an integer
	>>> bg.integer_validator("age", 0)
	Traceback (innermost last):
	ValueError: age must be greater than 0
	>>> bg.integer_validator("distance", -4)
	Traceback (innermost last):
	ValueError: distance must be greater than 0
	>>>
	>>>
	>>>
	>>>
	>>>
	>>>
	>>>
	>>>
	>>>
	>>>
	>>>
	>>>
	>>>
	>>>
	>>>
	>>>
